### PR TITLE
New Service: NEODyS website

### PR DESCRIPTION
**Pull request by KatKiker**
This adds a new service for accessing data from the objects listed on the NEODyS website (https://newton.spacedys.com/neodys/). 

I included a few simple tests, and a doc example in astroquery/docs/neodys. Let me know if anything else is needed.

### CODE REVIEW

**Comment by KatKiker** [comment]
Related to issue #1075 

Hopefully ready for review. Let us know if there's anything else that should be added.

**Review by keflavich — CHANGES_REQUESTED** [review]
Mostly looks good, but the `requests` need to be made using astroquery's internal request module so that appropriate headers are sent, and to enable other features.

File: astroquery/neodys/core.py
Comment (KatKiker) [reply astroquery/neodys/core.py]: Changes made! Thanks.

File: astroquery/neodys/core.py
Comment (KatKiker) [reply astroquery/neodys/core.py]: Changes made! Thanks.

**Review by keflavich — APPROVED** [review]

File: astroquery/neodys/core.py
Comment (bsipocz) [inline astroquery/neodys/core.py:14]: If you don't do any customisation this can be removed

File: astroquery/neodys/core.py
Comment (bsipocz) [inline astroquery/neodys/core.py:17]: Maybe make the optional ones kwarg only.
```suggestion
    def query_object(self, object_id, *, orbital_element_type="eq", epoch_near_present=0):
```

File: astroquery/neodys/core.py
Comment (bsipocz) [inline astroquery/neodys/core.py:83]: Is there any case when the response is not the expected result? E.g. `response.raise_for_status()` would only raise for HTTP Errors, but would let through any query related errors (as those are traditionally wrapped into a valid response), e.g. empty returns, etc.

File: astroquery/neodys/tests/test_neodys.py
Comment (bsipocz) [inline astroquery/neodys/tests/test_neodys.py:39]: Use some float comparison here and below, e.g. `assert_allclose`, as this produces a failure atm.

```
__________________________________________________ test_neodys_query ___________________________________________________

    def test_neodys_query():
        res = neodys.core.NEODyS.query_object(
            "2018VP1", orbital_element_type='eq', epoch_near_present=0)
>       assert res['Equinoctial State Vector'] == [1.5881497559198392,
        -0.037761493287362, 0.428349632624304, 0.018136658553998,
        0.021742631955829, 14.9386830433394]
E       assert [1.5881497559...9386830433043] == [1.5881497559...9386830433394]
E         At index 0 diff: 1.5881497559213003 != 1.5881497559198392
E         Use -v to get the full diff

astroquery/neodys/tests/test_neodys.py:39: AssertionError
```

File: docs/neodys/neodys.rst
Comment (bsipocz) [inline docs/neodys/neodys.rst:1]: Now we do test the code examples in the narrative docs. Please remove this line, and use the `.. doctest-remote-data::` directive instead of `.. code-block:: python` for the code blocks. Some tweaking may be required, but IMO it's totally worth it if the result is a tested documentation.

**Review by bsipocz — CHANGES_REQUESTED** [review]
Unfortunately, it took a while to get back to it, but overall it's in very good shape, so we can turn around it quickly. As time passed, many things changed in the main repo, so please rebase to pick up the newer CI configs, etc.

Then, please add a changelog entry. 

Also, I wonder whether this should go directly into the `solarsystem` namespace, at least it should have an alias there as the rest of the solar system modules.

Also, add the narrative docs link into `index.rst` similarly how the other modules do it.

**Comment by bsipocz** [comment]
@KatKiker - Could you please let us know the plans for this? Unfortunately, the maintainers can't take over this PR and finish it as it was opened from an org fork (and unlike to individual forks, we cannot push back to branches on org forks).

**Comment by KatKiker** [comment]
@bsipocz Thanks for getting back to this! I can resubmit the pull request from an individual fork, if that would help. I can also possibly work on some of the items myself, but I may not have the bandwidth to get to it for a several weeks. It's been a while since I worked on this and I'd have to refresh my memory on it. Let me know what would be best.

**Comment by bsipocz** [comment]
@KatKiker - Whichever approach works best for you will work for me (I'm happy to finish this one as there are only minor things to fix in a new PR, or equally happy to wait until you have some bandwidth to follow-up on the review in this PR).

**Comment by KatKiker** [comment]
@bsipocz Thanks. I went ahead and created a new PR #2618 with the fork from my individual account. Let me know if that works for you or if there's anything else I need to add/change. Excited to see this possibly integrated.

**Comment by bsipocz** [comment]
Closing this in favour of #2618

**Pull request closed without merging**
